### PR TITLE
Validate all the things

### PR DESCRIPTION
**Pull request by DanielWaterworth**

### CODE REVIEW

File: lib/pusher.js [hunk 1]
```diff
@@ -79,12 +94,9 @@ Pusher.forCluster = function(cluster, options) {
  * @returns {String} authentication signature
  */
 Pusher.prototype.authenticate = function(socketId, channel, data) {
-  if (typeof socketId !== "string" || socketId === "") {
-    throw new Error("Invalid socket id: '" + socketId + "'");
-  }
-  if (typeof channel !== "string" || channel === "") {
-    throw new Error("Invalid channel name: '" + channel + "'");
-  }
+  validateSocketId(socketId);
```
Comment (leggetter) [inline lib/pusher.js]: Could we please have some test cases for validating the socket_id?

File: lib/pusher.js [hunk 2]
```diff
@@ -10,6 +10,21 @@ var Config = require('./config');
...
+  if (typeof channel !== "string" || channel === "" || channel.match(/[^A-Za-z0-9_\-=@,.;]/)) {
+    throw new Error("Invalid channel name: '" + channel + "'");
+  }
+  if (channel.length > 200) {
+    throw new Error("Channel name too long: '" + channel + "'");
+  }
+}
+
+var validateSocketId = function(socketId) {
+  if (typeof socketId !== "string" || socketId === "" || !socketId.match(/^\d+\.\d+$/)) {
```
Comment (leggetter) [inline lib/pusher.js]: Is @zimbatm's comment about the required regexp [here](https://github.com/pusher/pusher-http-python/commit/afaf75d271ad447df215e9b32448f11b9565792a#commitcomment-11131227) relevant?

Comment (DanielWaterworth) [inline lib/pusher.js]: `\A` and `\Z` aren't valid in JS. `^` and `$` work the same, except in multiline mode. I have checked this.

Comment (leggetter) [inline lib/pusher.js]: :+1:

File: lib/pusher.js (hunk 1 above)
Comment (DanielWaterworth) [inline lib/pusher.js]: Done.

Comment (leggetter) [inline lib/pusher.js]: :+1:

**Pull request merged**